features/steps/validation_handling.py

```diff
@@ -0,0 +1,281 @@
+                    step_results = list(fn(context, inst = inst, **kwargs)) # note that 'inst' has to be a keyword argument
+                    for result in step_results:
+                        try:
+                            inst = inst.to_string()
+                        except AttributeError:  # AttributeError: 'entity_instance' object has no attribute 'to_string'
+                            inst = str(inst)
+
+                        validation_outcome = StepOutcome(inst=activation_inst, context=context, **result.as_dict())
+
+                        validation_outcome = ValidationOutcome(
+                            outcome_code=getattr(ValidationOutcomeCode, validation_outcome.outcome_code),
+                            # observed = validation_outcome.model_dump_json(exclude=('context', 'outcome_code'), exclude_none=True), #TODO (parse it correctly)
+                            # expected = validation_outcome.model_dump_json(exclude=('context', 'outcome_code'), exclude_none=True), #TODO (parse it correctly)
+                            observed=validation_outcome.model_dump(include=('observed'))["observed"],  # TODO (parse it correctly)
+                            expected=validation_outcome.model_dump(include=('expected'))["expected"],  # TODO (parse it correctly)
+                            feature=context.feature.name,
+                            feature_version=misc.define_feature_version(context),
+                            severity=getattr(OutcomeSeverity, "WARNING" if any(tag.lower() == "warning" for tag in context.feature.tags) else "ERROR"),
+                            check_execution_id=check_execution_id
+                        )
+                        context.gherkin_outcomes.append(validation_outcome)
+
+                    if not step_results:
+
+                        if isinstance(inst, (tuple, list)): # TODO -> this is quite dirty temp solution. Done because @given('Its attribute {attribute}') return a tuple, not an instance
+                            inst = inst[0]
+
+                        StepOutcome(inst=activation_inst,
+                                    context=context,
+                                    expected=None,
+                                    observed=None)  # expected / observed equal on passed rule?
+
+                        validation_outcome = ValidationOutcome(
+                            outcome_code=ValidationOutcomeCode.P00010,  # "Rule passed"
+                            # observed = validation_outcome.model_dump_json(exclude=('context', 'outcome_code'), exclude_none=True), #TODO (parse it correctly)
+                            # expected = validation_outcome.model_dump_json(exclude=('context', 'outcome_code'), exclude_none=True), #TODO (parse it correctly)
+                            observed='',  # TODO (parse it correctly)
+                            expected='',  # TODO (parse it correctly)
```

Comment:
```suggestion
                            observed=None,
                            expected=None,
```

This is the PASSED result am I right? So observed and expected is always null.

Reply:
Yes, applying.